### ISSUE TITLE
Implement a proper timeout per actor

### ISSUE THREAD

**Issue by cp2587** [wiremind/remoulade#65]
The current mechanism to kill an actor is not properly working leading to potential actor running indefinitely. This is very dangerous has potential deadlock are always a possibility and each actor should not have to implement it's own timeout/kill mechanism.  
How can we  improve the library so it can kill a long-running actor ?

**Comment by antoinerabany** [comment]
Yes. The issue is that a thread cannot be killed ([source](https://stackoverflow.com/questions/323972/is-there-any-way-to-kill-a-thread)). So we either need to use multiprocessing or libraries like greenlet maybe. I need to document more about those and how it is done in other librairies. But I'm alway open if you have any advice.

**Comment by cp2587** [comment]
yes, implement a lightwieht coroutine library: it will prove as effective as thread for I/O intensive task. If you use gevent, you need to have compliant code, that's why thread is better in that sense. In python 3, i am not sure gevent is still the canonical way to do it (since standard librairy offer async.io). We should think carefully.
@mehdigmira any thought ?
